pilot/pkg/serviceregistry/kube/controller/ambient/helpers.go

```diff
@@ -133,6 +133,10 @@ func IsPodRunning(pod *v1.Pod) bool {
 	return pod.Status.Phase == v1.PodRunning
 }
 
+func IsPodPending(pod *v1.Pod) bool {
+	return pod.Status.Phase == v1.PodPending
```

Comment:
Can we look at a more granular state than the `Phase`? To quote the docs:

> The phase of a Pod is a simple, high-level summary of where the Pod is in its lifecycle. The phase is not intended to be a comprehensive rollup of observations of container or Pod state, nor is it intended to be a comprehensive state machine.


Can we instead use the PodCondition: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions?

Specifically, `PodScheduled` and/or `PodReadyToStartContainers`

(what we really want here is `PodReadyToStartContainers` I think, as it guarantees network readiness, but that is currently a k8s beta feature - and since we have a CNI plugin strictly speaking we don't have to wait on k8s to tell us about network readiness)